### PR TITLE
Mailer update

### PR DESCRIPTION
**Pull request by gabriel-glo**
# Description

Allow adding template tags to other fields in Mailer settings (destination email, subject, confirmation email, .etc)

# Screenshots / Videos

image

### CODE REVIEW

**Review by mbmjertan — APPROVED** [review]
Looks good to me, but it would be great to add tests for this. (I can do that as well if you'd like)

File: tests/Mailer/MailerSettingsTest.php [hunk 1]
```diff
@@ -29,3 +29,17 @@
 	$this->assertSame(10, \has_filter(SettingsMailer::FILTER_SETTINGS_NAME, 'EightshiftForms\Mailer\SettingsMailer->getSettingsData()'), 'The callback getSettingsData should be hooked to custom filter hook with priority 10.');
 	$this->assertSame(10, \has_filter(SettingsMailer::FILTER_SETTINGS_IS_VALID_NAME, 'EightshiftForms\Mailer\SettingsMailer->isSettingsValid()'), 'The callback isSettingsValid should be hooked to custom filter hook with priority 10.');
 });
+
+test('isSettingsValid method returns correct values', function () {
+	$this->assertSame(true, $this->mailerSettings->isSettingsValid('1234'));
```
Comment (dingo-d) [inline tests/Mailer/MailerSettingsTest.php:34]:
```suggestion
	expect($this->mailerSettings->isSettingsValid('1234'))->toBeTrue();
```

File: tests/Mailer/MailerSettingsTest.php [hunk 2]
```diff
@@ -29,3 +29,17 @@
...
+});
+
+test('getSettingsSidebar method returns correct values', function () {
+	expect($this->mailerSettings->getSettingsSidebar())->toBeArray();
+	expect($this->mailerSettings->getSettingsSidebar())->toHaveKeys(['label', 'value', 'icon']);
```
Comment (dingo-d) [inline tests/Mailer/MailerSettingsTest.php:39]: Expectations are chainable

```suggestion
	expect($this->mailerSettings->getSettingsSidebar())
		->toBeArray()
		->toHaveKeys(['label', 'value', 'icon']);
```

😉

File: tests/Mailer/MailerSettingsTest.php [hunk 3]
```diff
@@ -29,3 +29,17 @@
 	$this->assertSame(10, \has_filter(SettingsMailer::FILTER_SETTINGS_NAME, 'EightshiftForms\Mailer\SettingsMailer->getSettingsData()'), 'The callback getSettingsData should be hooked to custom filter hook with priority 10.');
 	$this->assertSame(10, \has_filter(SettingsMailer::FILTER_SETTINGS_IS_VALID_NAME, 'EightshiftForms\Mailer\SettingsMailer->isSettingsValid()'), 'The callback isSettingsValid should be hooked to custom filter hook with priority 10.');
```
Comment (dingo-d) [inline tests/Mailer/MailerSettingsTest.php:30]: Can you convert this to expectations API? 🙏🏼

**Review by dingo-d — CHANGES_REQUESTED** [review]
Expectations api replacement

**Review by iruzevic — APPROVED** [review]
Feature looks good, fix Test comments

**Review by dingo-d — APPROVED** [review]

**Review by dadadavorin — APPROVED** [review]

**Pull request merged**